packages/contracts-bedrock/src/dispute/FaultDisputeGameN.sol

```diff
@@ -340,13 +362,13 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
         // known, we can compute the next position by moving left or right depending on whether
         // or not the move is an attack or defense.
         Position parentPos = parent.position;
-        Position nextPosition = parentPos.move(_isAttack);
+        Position nextPosition = parentPos.moveN(N_BITS, _attackBranch);
         uint256 nextPositionDepth = nextPosition.depth();
 
         // INVARIANT: A defense can never be made against the root claim of either the output root game or any
         //            of the execution trace bisection subgames. This is because the root claim commits to the
         //            entire state. Therefore, the only valid defense is to do nothing if it is agreed with.
-        if ((_challengeIndex == 0 || nextPositionDepth == SPLIT_DEPTH + 2) && !_isAttack) {
+        if ((_challengeIndex == 0 || nextPositionDepth == SPLIT_DEPTH + 2 * N_BITS) && 0 != _attackBranch) {
```

Comment:
Should the condition be `3 != _attackBranch` ?

Reply:
From my understanding.
1. should be `attachBranch !=0`, because the root claim (which only lies in the first branch) can't be defended
2. because the [depth definition](https://perfect-amphibian-929.notion.site/Multi-Section-Fault-Proof-Explained-13c3474092da46218a51835fd51b2e63) using 2-ary depth

Reply:
The root is that the root claim of a block transition is a sub claim of a block-level-output claim at depth `SPLIT_DEPTH`.  Since a block-level-output claim at depth `SPLIT_DEPTH` has only one sub-claim, i.e., the root claim, the following claims at SPLIT_DEPTH + 2 can only attack the root claim (_attackBranch must be 0).